### PR TITLE
Desert region mod

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Content "A desert region mod"```

#### Purpose of change
A collaboration with @DracoGriffin who will take over from this ( the monsters added are from their branch of my original branch ) and Draco will open it up to the community to add stuff to it, I'm just trying to lay the groundwork for some stuff that makes other regions more possible.

#### Describe the solution
Using the features of the regional settings, and some added code to weather_gen to allow more fine control of weather variations. There is now a desert-ish climate, can get very cold at night, but obviously very hot during the day.
The generation of forests and swamps and buildings and so on require some tweaking, but I've added a new default_oter of "desert" which has scorched earth and sand as groundcover and cactii, and an example cactus fruit from that.
I've tried to add a scaling mechanism to the rivers so they are drier and patchier, it looks a bit crap, and to be honest, that could probably be removed, as deserty regions still have rivers, but I thought that the struggle to find water would make it interesting.

#### Describe alternatives you've considered
Removing the river-scaling code, but it may be salvageable, ralreegorganon said that the current algorithm does not suit scaling, so it may be best removed until a new algorithm is developed.

#### Additional context
This is a mod thats a test mod of sorts, that as I said @DracoGriffin wants to work on, and in its current state is not ready for a full game experience, but is ready to open to the community for testing and improvement and hopefully will lead to interest in other types of regions and code to support that.

image

image

### CODE REVIEW

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by nexusmrsep** [comment]
You may want to check if changing some `game_constants.h` would fit your mod. Most of them are not JSONized, so this infrastructural change might be also needed.

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by kevingranade** [comment]
This got caught out by the mandatory description change.

File: data/mods/desert_region/desert_regional_map_settings.json [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+    "type": "region_settings",
+    "id": "default",
+    "default_oter": "desert",
+    "default_groundcover": [ [ "t_searth_test", 4 ], [ "t_sand", 1 ] ],
+    "river_scale": 0.2,
+    "field_coverage": {
+      "percent_coverage": 0.1333,
+      "default_ter": "t_shrub",
+      "other": {
+        "f_cholla_cactus": 12.3333,
```
Comment (kevingranade) [inline data/mods/desert_region/desert_regional_map_settings.json]:
```suggestion
        "f_cactus": 12.3333,
```

File: data/mods/desert_region/desert_terrain.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    "symbol": ".",
+    "color": "light_gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "DIGGABLE", "FLAT" ],
+    "bash": { "sound": "thump", "ter_set": "t_null", "str_min": 50, "str_max": 100, "str_min_supported": 100, "bash_below": true }
+  },
+  {
+    "type": "terrain",
+    "id": "t_cactus",
+    "name": "cactus",
```
Comment (kevingranade) [inline data/mods/desert_region/desert_terrain.json]:
```suggestion
    "name": "cactus",
    "description": "A tall cactus.",
```

**Comment by kevingranade** [comment]
Looks good to go other than the two load errors.

**Comment by adamkad1** [comment]
is there a mod to add random overmaps to be deserts? unless this is, it kinda looks like iit makes the whole world a desert

**Pull request merged**